circuit_knitting_toolbox/circuit_cutting/cutting_reconstruction.py

```diff
@@ -13,13 +13,93 @@
 
 from __future__ import annotations
 
+from collections.abc import Sequence
+
 import numpy as np
+from qiskit.quantum_info import PauliList
+from qiskit.result import QuasiDistribution
 
-from ..utils.observable_grouping import CommutingObservableGroup
+from ..utils.observable_grouping import CommutingObservableGroup, ObservableCollection
 from ..utils.bitwise import bit_count
+from .cutting_decomposition import decompose_observables
+from .qpd import WeightType
+
+
+def reconstruct_expectation_values(
+    quasi_dists: Sequence[Sequence[Sequence[tuple[QuasiDistribution, int]]]],
+    coefficients: Sequence[tuple[float, WeightType]],
+    observables: PauliList | dict[str | int, PauliList],
```

Comment:
fyi, we'll need both `observables` and `subobservables` here to support `SparsePauliOp`.

Reply:
I said earlier that we will need both `subobservables` and `observables` for reconstruction with `SparsePauliOp`, but this is not sufficient, I am now thinking.  We actually need `partition_labels` again, too, so that we know how each observable matches up with the different subobservables.  But at that point, we can re-derive the `subobservables` from `observables` and `partition_labels`.  (Of course, if the circuit is not separated, we don't actually need `partition_labels` because it is trivial.)

Reply:
OK what do you suggest we do for 0.2.0? Just hold it as it is for now and rename this to `subobservables`?

Reply:
For `execute`, I think it makes sense to use `subobservables`.  For reconstruction, I think `observables` actually makes more sense.